internal/provider/resource_pipeline.go

```diff
@@ -0,0 +1,366 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/humanitec/humanitec-go-autogen"
+	"github.com/humanitec/humanitec-go-autogen/client"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &ResourcePipeline{}
+var _ resource.ResourceWithImportState = &ResourcePipeline{}
+
+func NewResourcePipeline() resource.Resource {
+	return &ResourcePipeline{}
+}
+
+// ResourceRule defines the resource implementation.
+type ResourcePipeline struct {
+	client *humanitec.Client
+	orgID  string
+}
+
+func (r *ResourcePipeline) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_pipeline"
+}
+
+func (r *ResourcePipeline) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "An Automation Rule defining how and when artefacts in an environment should be updated.",
```

Comment:
Unrelated to this line, but could you also run a `go generate ./...` to update the docs?

You might also need to add a file similar to https://github.com/humanitec/terraform-provider-humanitec/blob/main/examples/resources/humanitec_resource_definition/import.sh if the resource supports importing.

Reply:
Done!